src/main/java/WayofTime/bloodmagic/api/ItemStackWrapper.java

```diff
@@ -13,17 +14,18 @@
     public final Item item;
     public final int stackSize;
     public final int meta;
+    public final NBTTagCompound nbtTag;
 
     public ItemStackWrapper(Item item, int stackSize) {
```

Comment:
Remove @ReqiredArgsConstructor and make your own 

    public ItemStackWrapper(Item item, int stackSize, int meta) {
        // Blah
    }

so that NBT isn't set by constructor. Use a Setter.